app/org/sagebionetworks/bridge/services/AuthenticationServiceImpl.java

```diff
@@ -161,7 +162,13 @@ public void signUp(Study study, SignUp signUp, boolean sendEmail) {
                 throw new StudyLimitExceededException(study);
             }
             accountDao.signUp(study, signUp, sendEmail);
-
+        } catch(EntityAlreadyExistsException eae) {
+            // Do not indicate to the end user that this email address has already been taken, as this 
+            // leaks information about who is in the study. Instead send a reset password request to the 
+            // email address in case user has forgotten password and is trying to sign up again.
+            logger.info(String.format("Sign up attempt for existing account: %s, %s", study.getIdentifier(), signUp.getEmail()));
+            Email email = new Email(study.getIdentifier(), signUp.getEmail());
+            requestResetPassword(study, email);
```

Comment:
Yes, the goal is that it look 100% like they succeeded. I admit we are technically lying when we return a 201!